### PR TITLE
refactor: workspace loading logic

### PR DESCRIPTION
**Pull request by Himself65**
Fixes: https://github.com/toeverything/AFFiNE/issues/1952

Related: https://github.com/toeverything/AFFiNE/issues/1810

### CODE REVIEW

File: apps/web/src/atoms/root.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+ * Fetch all workspaces from the Plugin CRUD
+ */
+export const workspacesAtom = atom<Promise<AllWorkspace[]>>(async get => {
+  const flavours: string[] = Object.values(WorkspacePlugins).map(
+    plugin => plugin.flavour
+  );
+  const jotaiWorkspaces = get(rootWorkspacesMetadataAtom).filter(workspace =>
+    flavours.includes(workspace.flavour)
+  );
+  const workspaces = await Promise.all(
```
Comment (pengx17) [inline apps/web/src/atoms/root.ts:25]: I think we do not actually need to wait for whole workspaces to load. We just need the workspace id and then load individually.

File: apps/web/src/atoms/index.ts [hunk 2]
```diff
@@ -1,40 +1,68 @@
...
-export const currentWorkspaceIdAtom = atom<string | null>(null);
-export const currentPageIdAtom = atom<string | null>(null);
-export const currentEditorAtom = atom<Readonly<EditorContainer> | null>(null);
+/**
+ * @deprecated Use `rootCurrentWorkspaceIdAtom` directly instead.
+ */
+export const currentWorkspaceIdAtom = rootCurrentWorkspaceIdAtom;
+
+// todo(himself65): move this to the workspace package
+rootWorkspacesMetadataAtom.onMount = setAtom => {
```
Comment (pengx17) [inline apps/web/src/atoms/index.ts:24]: I am not sure about onMount, does it run one every use cases? Which means the creatFirst could run every place that is being used?

Comment (himself65) [reply apps/web/src/atoms/index.ts:24]: When there's no workspace, we should call `app:init` event

File: apps/web/src/atoms/root.ts (hunk 1 above)
Comment (himself65) [reply apps/web/src/atoms/root.ts:25]: image

For now there is only modals needs to read all workspaces. Not a big deal

**Pull request merged**